ValueExtractor has some edge cases breaking IL2CPP builds

### Environment

* Code source: Package manager
* Code version: "io.extendreality.zinnia.unity": "1.16.0"
* Platform version: Unity 2019.2.21f1
* 3rd party dependencies: None

### Steps to reproduce

- Create a new project,
- Add a script to the main scene that does the following
```csharp
using UnityEngine;
using Zinnia.Data.Operation.Extraction;

public class Example : MonoBehaviour
{
    void Start()
    {
        // We force the type to appear
        Debug.Log(typeof(SurfaceDataCollisionPointExtractor));
    }
}
```
- Save the scene
- Set scripting backend to IL2CPP and build

### Expected behavior

Should build.

### Current behavior

IL2CPP compilation error.

It seems that `ValueExtractor` has some edge cases that breaks the AOT scripting restrictions causing the build to fail.

I tried setting Stripping level to high and it still won't build.

Also, the class that I used to reproduce the error is `SurfaceDataCollisionPointExtractor`, but some other classes, like `SurfaceDataCollisionDataExtractor` don't cause this problem, so it seems to be a problem with the type `RaycastHit`.

Part of the log which seems useful says: 
> error C2440: 'type cast': cannot convert from 'int' to RaycastHit_t19695F18F9265FE5425062BBA6A4D330480538C3'

@thestonefox I tried it and still have the same problem

> error C2440: 'type cast': cannot convert from 'int' to 'RaycastHit_t19695F18F9265FE5425062BBA6A4D330480538C3'

--- a/Runtime/Data/Operation/Extraction/ValueExtractor.cs
+++ b/Runtime/Data/Operation/Extraction/ValueExtractor.cs
@@ -20,7 +20,7 @@
         /// <summary>
         /// The source to extract from.
         /// </summary>
-        [Serialized, Cleared]
+        [Serialized]
         [field: DocumentedByXml]
         public TSourceElement Source { get; set; }
 

